internal/declarative/v2/reconciler.go

```diff
@@ -316,40 +316,52 @@ func hasDiff(oldResources []Resource, newResources []Resource) bool {
 }
 
 func (r *Reconciler) checkTargetReadiness(
-	ctx context.Context, clnt Client, obj Object, target []*resource.Info,
+	ctx context.Context, clnt Client, manifest Object, target []*resource.Info,
 ) error {
-	status := obj.GetStatus()
+	status := manifest.GetStatus()
 
 	resourceReadyCheck := r.CustomReadyCheck
 
-	state, err := resourceReadyCheck.Run(ctx, clnt, obj, target)
+	crStateInfo, err := resourceReadyCheck.Run(ctx, clnt, manifest, target)
 
-	if errors.Is(err, ErrResourcesNotReady) || errors.Is(err, ErrCustomResourceStateNotFound) ||
-		errors.Is(err, ErrDeploymentNotReady) {
-		waitingMsg := fmt.Sprintf("waiting for resources to become ready: %s", err.Error())
-		r.Event(obj, "Normal", "ResourceReadyCheck", waitingMsg)
-		obj.SetStatus(status.WithState(StateProcessing).WithOperation(waitingMsg))
-		return err
+	if crStateInfo.State == StateProcessing {
```

Comment:
is it ok to check the state before error check?

Reply:
The position is irrelevant, if you look at those returned `crStateInfo`, if the `State` is `StateProcessing`, the returned `error` is always `nil`, but same as previous comments, let's deal with error first to keep consistent.

Reply:
:+1: You are right, that by the **current** implementation it doesn't matter. But implementation will be for sure changed in the future. The `ReadyCheck` interface doesn't declare any specific error-handling policy for the `Run` method. That's why we should stick to the default, idiomatic golang error handling: If the returned error != nil, we ignore any returned values.
If  it's necessary for the `Run` method to have a custom error handling policy, it should be specified somehow (in the godoc?)